content/v2.1/helpers/overview.md

```diff
@@ -3,8 +3,231 @@ title: Overview
+
+### Usage
+
+Here is how you can use it **in templates**:
+
+```ruby
+<%= tag.div(id: "el") do %>
+  <p>Template content can be mixed in.</p>
+  <%= tag.p("Also nested tag builders.") %>
+<% end %>
+```
+
+This will render:
+
+```html
+<div id="el">
+  <p>Template content can be mixed in.</p>
+  <p>Also nested tag builders.</p>
+</div>
+```
+
+[In parts](/v2.1/views/parts), you can access your helpers via the `helpers` object.
+
+Given you have a view exposure defined:
+
+```ruby
+# app/views/books/show.rb
+
+module Bookshelf
+  module Views
+    module Books
+      class Show < Bookshelf::View
+        expose :fancy_header do
+          "Fancy header"
+        end
+      end
+    end
+  end
+end
+```
+
+Then you can have the part with the helper used.
+
+```ruby
+# app/views/parts/fancy_header.rb
+
+module Bookshelf
+  module Views
+    module Parts
+      module FancyHeader
+        def content
+          helpers.tag.h1(value)
+        end
+      end
+    end
+  end
+end
+```
+
+Then in the template you can just access the part:
+
+```ruby
+# app/templates/books/show.html.erb
+
+<%= fancy_header %>
+```
+
+### Features
+
+Here are lists of features with the corresponding examples:
+
+**Auto-closing tags according to HTML5 spec**
+
+```ruby
+tag.div # => <div></div>
+```
+
+**Content string as first argument**
+
+```ruby
+tag.img # => <img>
+tag.div("First argument is content")        # => <div>First argument is content</div>
+```
+
+**Accepts content as block returning a string**
+
+```ruby
+tag.div { "Content in the block" }     # => <div>Content in the block</div>
+```
+
+**Allows for nesting tags via argument or block**
+
+```ruby
+tag.div(tag.p("Nested tag builder")) # => <div><p>Nested tag builder</p></div>
+
+tag.div do
+  tag.p("Also nested tag builders.")
+end
+```
+
+**It builds attributes from given hash**
+
+```ruby
+tag.div("With Additional attributes", class: ["a", "b"]) # => <div class="a b">With additional attributes</div>
+```
+
+**Builds nested HTML attributes attributes from given hash**
+
+```ruby
+tag.div(id: "el", data: {x: "y"}) # => <div id="el" data-x="y"></div>
+tag.div(id: "el", aria: {x: "y"}) # => <div id="el" aria-x="y"></div>
+```
+
+**Allows for dynamic attributes visibility control**
+
+```ruby
+tag.div("Dynamic attributes", class: {"a": true, "b": false}) # => <div class="a">Dynamic Attributes</div>
+```
+
+**Supports custom tags**
+
+```ruby
+tag.custom_tag("hello") # => <custom-tag>hello</custom-tag>
+```
+
+
+### Escaping HTML input
+
+The tag contents are automatically escaped for security reasons:
+
+```ruby
+tag.p("<script>alert()</script>")         # => <p>&lt;script&gt;alert()&lt;/script&gt;</p>
+tag.p(class: "<script>alert()</script>")  # => <p class="&lt;script&gt;alert()&lt;/script&gt;"></p>
+```
+
+To bypass it use `html_safe` on the given content
+
+```ruby
+tag.p("<em>safe content</em>".html_safe)  # => <p><em>safe content</em></p>
+```
+
+### Standalone use hanami-view
```

Comment:
@swilgosz I'm for keeping out of the Guides any standalone usage of the Hanami gems.

Guides -> full Hanami app.
Docs -> standalone gems

Reply:
Removed!